File: code/backend/foodometer-order-service/src/main/java/com/springmicro/foodometer/document/Address.java
package com.springmicro.foodometer.document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "Address")
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Address {
    @Id
    private String id;
    private String line1;
    private String line2;
    private String line3;
    private String state;
    private String city;
    private String pinCode;
}

File: code/frontend/food-o-gui/src/app/_model/address.ts
export class Address {
    id: string;
    line1: string;
    line2: string;
    line3: string;
    state: string;
    city: string;
    pinCode: string;

    constructor(id?: string,
        line1?: string,
        line2?: string,
        line3?: string,
        state?: string,
        city?: string,
        pinCode?: string) {
        this.id = id;
        this.line1 = line1;
        this.line2 = line2;
        this.line3 = line3;
        this.state = state;
        this.city = city;
        this.pinCode = pinCode;
    }
}
